downloader/src/main/kotlin/Downloader.kt

```diff
@@ -270,8 +270,13 @@ class Downloader(private val config: DownloaderConfiguration) {
             return RepositoryProvenance(pkg.vcsProcessed, pkg.vcsProcessed.revision)
         }
 
+        val vcsConfig = config.getVersionControlSystemConfiguration(applicableVcs.type)
```

Comment:
Commit message:

> (e.g., `revision`, `recursive`)

`revision` probably is not such a good example as it usually changes on a call-by-call basis. Better use `priority` as another example, as that can be static per ORT run.

Reply:
Agreed. I convert the PR back to draft and create a solution based on TypedConfigurablePluginFactory, like e.g. ScannerWrapperFactory does.